ledger/sandbox/src/main/resources/db/migration/postgres/V24__Stable_offsets_archival.sql

```diff
@@ -0,0 +1,10 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V24: Stable offsets archival
+--
+-- create_consumed_at is stored as byte array
+---------------------------------------------------------------------------------------------------
+
+ALTER TABLE participant_events ALTER COLUMN create_consumed_at TYPE bytea using decode(create_consumed_at, 'hex');
```

Comment:
Hm, considering that this always has been `varchar` and I missed to updated it in V21, the database can now potentially contain decimal based offsets as well as `bytea` automatically converted to `varchar`, right?

Reply:
Probably, how can we deal with those?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5371/commits/7562dbdbbf8e90b25a95e481717946e2fcc4bab6

Reply:
As noted offline, since v20 and v21 migrations landed in the same release, this cannot have occurred. Furthermore, byte arrays casted to varchars can safely be casted back to byte arrays. Addressed in https://github.com/digital-asset/daml/pull/5371/commits/de983641b6d7521e8f5df214919c9e96c7c97912.